Sources/CryptorRSA/CryptorRSAKey.swift

```diff
@@ -318,28 +318,41 @@ extension CryptorRSA {
 				
 				throw Error(code: ERR_CREATE_CERT_FAILED, reason: "Unable to create certificate from certificate data.")
 			}
+			
+			var key: SecKey? = nil
+			#if swift(>=4.2)
+			if #available(macOS 10.14, iOS 12.0, watchOS 5.0, *) {
+				
+				key = SecCertificateCopyKey(certData)
+				
+			} 
+			#endif
+			if key == nil {
 		
 			#if os(macOS)
 		
 				// Now extract the public key from it...
-				var key: SecKey? = nil
 				let status: OSStatus = withUnsafeMutablePointer(to: &key) { ptr in
 					
 					// Retrieves the public key from a certificate...
 					SecCertificateCopyPublicKey(certData, UnsafeMutablePointer(ptr))
 				}
-				if status != errSecSuccess || key == nil {
+				if status != errSecSuccess {
 					
 					throw Error(code: ERR_EXTRACT_PUBLIC_KEY_FAILED, reason: "Unable to extract public key from data.")
 				}
 		
 			#else
 		
-				let key = SecCertificateCopyPublicKey(certData)
+				key = SecCertificateCopyPublicKey(certData)
 		
 			#endif
+			}
 		
-			return PublicKey(with: key!)
+		guard let createdKey = key else {
```

Comment:
Why not a simple _if statement_?  It's actually _faster_ to execute than the _guard_ statement (or at least it used to be from data I've read in the past).

Reply:
guard is a language feature and I believe the compiler optimizes it into an if anyway. Avoiding force unwraps is generally good since even if it will always unwrap at the moment, it makes the code more robust in case the code above changes.